daemons/ipa-kdb/ipa_kdb_principals.c

```diff
@@ -351,10 +342,19 @@ static enum ipadb_user_auth ipadb_get_user_auth(struct ipadb_context *ipactx,
     if (ua == IPADB_USER_AUTH_NONE)
         ua = gua;
 
+    /* If no mechanisms are set, allow every auth method for now. */
+    if (ua == IPADB_USER_AUTH_NONE) {
+        ua = IPADB_USER_AUTH_PASSWORD
+           | IPADB_USER_AUTH_RADIUS
+           | IPADB_USER_AUTH_OTP
+           | IPADB_USER_AUTH_PKINIT
+           | IPADB_USER_AUTH_REINFORCED
+           ;
+    }
+
```

Comment:
This is prone to error on following changes, I think we have two options here:
1) do not do this and simply return kerr=0 if ua == IPADB_USER_AUTH_NONE in ipa_kdcpolicy_check_as() befoire we even go through auth_indicators array and save some computing time (preferred).
2) add a macro IPADB_USER_AUTH_ALL in the headers where all other IPADB_USER_AUTH_* defines are so that if someone adds a new one they will see it is needed in there and not need to check all places in the code to figure out there is one where all mechanisms need to be explicitly listed.

I favor (2) less because it is still error prone and it will always be the case that if no indicators are set we accept any pre-auth mechanism so we should just bail out of policy checking right away, and not cycle through each and every one we defined (also it will be faster that way).

Reply:
Can you clarify your preference?  You state that you favor 2, but 1 is preferred?

Reply:
Went for (1)

Reply:
err sorry I meant I disfavor (2) ... sigh... language, thanks @tengcm2015 for reading my mind :-)